Lib/types.py

```diff
@@ -300,4 +300,81 @@ def wrapped(*args, **kwargs):
 NoneType = type(None)
 NotImplementedType = type(NotImplemented)
 
+class TracebackGroup:
+    def __init__(self, excs, frame):
+        self.tb_frame = frame
+        self.tb_next_map = {} # TODO: make it a weak key dict
+        for e in excs:
+            self.tb_next_map[e] = e.__traceback__
+
+class ExceptionGroup(BaseException):
+
+    def __init__(self, excs, frame=None):
+        self.excs = set(excs)
+        self.frame = frame or sys._getframe()
```

Comment:
A shame that the exception must keep the frame alive -- normally that is done through the traceback, so at least you can sever the link by setting `__traceback__ = None`.

(In case you wonder why I care, there is a painful bit of ancient history around cycles involving exceptions and frames dating back to the Python 1.x days. This is also why `except E as e` deletes `e` at the end of the handler block.)

Reply:
I've removed it, we didn't really need it but I was confused.